### ISSUE TITLE
Conversion to_dict method removes desiredState level

### ISSUE THREAD

**Issue by inf17101** [eclipse-ankaios/ank-sdk-rust#84]
The `to_dict` method of [complete_state.rs](https://github.com/eclipse-ankaios/ank-sdk-rust/blob/main/src/components/complete_state.rs#L233) converts from CompleteState object to a serde_yaml::Mapping and accidentially removes the `desiredState` nested level.

## Current Behavior

To_dict converts to a wrongly flattened state layout removing the "desiredState" nested level.
All state fields like `agents`, `workloads`, `configs` are on the same level. 

## Expected Behavior

Correct CompleteState layout in to_dict method.

## Steps to Reproduce

1. Create a CompleteState object from a Manifest file
2. Call `complete_state.to_dict()`
3. Try to access key "desiredState" which will return no value
4. Print the `complete_state_dict` and you will see that the level does not exist

## Context (Environment)

All supported platforms

## Additional Information

Perhaps the python sdk suffers from the same issue.

## Final result

**To be filled by the one closing the issue.**

### Tasks

- [ ] Assign yourself as `assignee`;
- [ ] Implement enhancement;
- [ ] Document the code similar to how it's done for the other code;
- [ ] Check and fix utests, coverage, clippy and doc; 
- [ ] Create Pull Request and fix possible workflow failures; 
- [ ] Add the `ready_for_review` label.

**Comment by GabyUnalaq** [comment]
The fix has been merged. This issue can be closed.
